model/project.go

```diff
@@ -1231,6 +1231,43 @@ func (p *Project) FindTaskGroup(name string) *TaskGroup {
 	return nil
 }
 
+// FindTaskGroupForTask returns a specific task group from a project
+// that contains the given task.
+func (p *Project) FindTaskGroupForTask(bvName, taskName string) *TaskGroup {
+	tgWithTask := map[string]TaskGroup{}
+	for _, tg := range p.TaskGroups {
+		for _, t := range tg.Tasks {
+			if t == taskName {
+				tgWithTask[tg.Name] = tg
+				fmt.Println("===", tg.Name, "===")
+			}
+		}
+	}
+
+	for _, bv := range p.BuildVariants {
+		if bv.Name != bvName {
+			continue
+		}
+		for _, t := range bv.Tasks {
+			// Check inline task groups.
+			if t.TaskGroup != nil {
+				for _, tg := range t.TaskGroup.Tasks {
+					if tg == taskName {
+						return t.TaskGroup
+					}
+				}
+				continue
+			}
+			// Check if the an already found task group contains the task.
+			if tg, ok := tgWithTask[t.Name]; ok {
```

Comment:
Rather than pre-populate the task groups containing the task, and then later using it after iterating through the BV task list, would it be clearer to instead first look for the BV of interest, and if the task is _not_ found in an inline task group within the BV, just `break` out of the BV loop? Then afterwards, it can fall back to iterating through the top-level task group list. Would make this possibly a little clearer rather than pre-iterating through the top-level task groups.